spec/models/searchgov_url_spec.rb

```diff
@@ -890,6 +890,75 @@
       end
     end
 
+    context 'when url has different canonical_url' do
+      let(:canonical_url) { 'https://www.agency.gov/new.html' }
+
+      before do
+        expect(I14yDocument).not_to receive(:create).
+          with(hash_including(title: 'My Title', description: 'My description'))
+        stub_request(:get, url).
+          to_return(status: 301,
+                    body: html,
+                    headers: { location: canonical_url })
+        stub_request(:get, canonical_url).
+          to_return(status: 200,
+                    body: html,
+                    headers: { content_type: 'text/html' })
+      end
+
+      it 'creates a new url' do
+        expect(described_class).to receive(:create).with(url: canonical_url)
+        fetch
+      end
+    end
+
+    context 'when canonical_url is created' do
+      let(:canonical_url) { 'https://www.agency.gov/canonical' }
+      let(:searchgov_document) { searchgov_url.searchgov_document }
+
+      before do
+        allow(searchgov_url).to receive(:document).and_return(searchgov_document)
+        allow(searchgov_document).to receive(:canonical_url).and_return(canonical_url)
+        allow(described_class).to receive(:persisted?).and_return(true)
+      end
+
+      it 'updates last_crawl_status with canonical URL info' do
+        expect { searchgov_url.send(:use_canonical_url) }.
+          to raise_error(SearchgovUrl::SearchgovUrlError, "Created canonical url #{canonical_url}")
+        expect(searchgov_url.last_crawl_status).to include("Canonical URL: #{canonical_url}")
+      end
+    end
+
+    describe '#check_canonical_url' do
```

Comment:
We shouldn't be testing private methods like this, and I'm not sure why `searchgov_document` is relevant here.  The remaining tests here will need to be rethought, but I don't want to go into too much detail until we see what the refactor of the `use_canonical_url` looks like.